src/include/network/services/routing.rb

```diff
@@ -91,19 +91,19 @@ def RoutingMainDialog
 
     def config
       # TODO: get it from some config holder
-      @routing_config ||= Yast::Lan.yast_config
+      @routing_config = Yast::Lan.yast_config
     end
 
     def routing_table
-      @routing_table_widget ||= Y2Network::Widgets::RoutingTable.new(config.routing.tables.first)
+      @routing_table_widget = Y2Network::Widgets::RoutingTable.new(config.routing.tables.first)
```

Comment:
hmm, I am not sure I like it as it basically mean that you always create new instance of widget. So if widget store itself any value, it will be forgotten.

Reply:
Do not like neither, so, what about caching the widget and just renew the cache in case the table changed?

Reply:
well, on one side we need to have widget same during dialog run. On other hand when dialog is shown again with different values, we need to reset it. That is a problem of includes in module which uses instance variable. I am fighting it a lot now when trying to abstract some widget outside. Not sure about proper solution now intermediate. Maybe add TODO or FIXME and use what you did in this PR with note in head that it cannot be the final solution.